### PR TITLE
Diags naming

### PR DESCRIPTION
**Pull request by painter1**
Output file names are now normally chosen in a central place, a new file form_filenames.py.  Graphics file names are the same as before.  NetCDF output file names are now very similar to the corresponding graphics file names.   I changed filenames in the automated test baselines so that the tests will all pass.  We still need more work on file names as well as titles and so forth; but this resolves the worst problems and will make it easier to work on file names in the future.

### CODE REVIEW

**Comment by mcenerney1** [comment]
All of the tests pass.

**Comment by mcenerney1** [comment]
While testing the name I found an issue. When running metadiags with
metadiags --package AMWG --set 5 --model path=$HOME/uvcmetrics_test_data/cam35_data/,climos=yes,name=junk --obs path=$HOME/uvcmetrics_test_data/obs_data/,climos=yes --outputdir $HOME/diagout/

All of them failed for example
DEBUG (from plotplan.py 312): now composite plot
INFO (from diags 351): --------------------------------- res is not none



STDOUT and STDERR

CDMS I/O error: Determining type of file /Users/mcenerney1/uvcmetrics_test_data/obs_data/NCEP.xml.orig; must specify dictionary (control) file
jfp
jfp running this python: /Users/mcenerney1/anaconda/bin/python
UV-CDAT Diagnostics, command-line version
/Users/mcenerney1/anaconda/bin/diags --model path=/Users/mcenerney1/uvcmetrics_test_data/cam35_data,climos=yes,type=model --obs path=/Users/mcenerney1//uvcmetrics_test_data/obs_data/,climos=yes,filter=f_startswith('NCEP') --logo no --colormaps diff=bl_to_darkred model=rainbow obs=rainbow --package AMWG --set 5 --seasons ANN --vars PSL --outputdir /Users/mcenerney1//diagout/amwg --prefix set5 --postfix NCEP --log_level DEBUG --log_file /Users/mcenerney1//diagout/amwg/DIAGS_OUTPUT/42e8208b93005d3fa6095699b9509e40/AMWG_5_NCEP_PSL_ANN.log
Traceback (most recent call last):
  File "/Users/mcenerney1/anaconda/bin/diags", line 762, in <module>
    run_diags(o)
  File "/Users/mcenerney1/anaconda/bin/diags", line 356, in run_diags
    postn=postname, region=r_fname, aux=[aux] )
  File "/Users/mcenerney1/anaconda/lib/python2.7/site-packages/metrics/frontend/form_filenames.py", line 133, in form_file_rootname
    fname = '_'.join( basen, season, '_'.join(vars), '_'.join(aux), postn )
TypeError: sequence item 0: expected string, NoneType found

**Comment by painter1** [comment]
I ran the same metadiags command, with slightly different paths, and did not get this sort of error.  I did get errors about certain requested variables not existing - that's basically ok, and anyway is addressed by another branch.  I also tried the specific diags command mentioned above, and it worked fine.

@mcenerney1, would you please try again?  But first run an "anaconda activate" so your Python is in a suitable UV-CDAT environment (anaconda/envs/<name>/bin/python), and reinstall uvcmetrics there.  Just running the above diags command should be enough to tell whether that helps.

File: src/python/frontend/diags.py [hunk 1]
```diff
@@ -9,10 +9,13 @@
 ### Look for speed improvements
 
 import hashlib, os, pickle, sys, os, time, re, pdb, logging
+print "jfp"
```
Comment (doutriaux1) [inline src/python/frontend/diags.py]: @painter1 no print please

File: src/python/packages/amwg/amwg.py [hunk 2]
```diff
@@ -4290,7 +4294,7 @@ def _results(self, newgrid=0):
         #pdb.set_trace()
         results = plot_plan._results(self, newgrid)
         if results is None:
-            print "WARNING, AMWG plot set 12 found nothing to plot"
+            print "WARNING, AMWG plot set 14 found nothing to plot"
```
Comment (doutriaux1) [inline src/python/packages/amwg/amwg.py]: @painter1 I know it's not you but please can you move this one to the logger as well

**Comment by painter1** [comment]
After a couple more builds I could reproduce the problem which @mcenerney1 reported.  More to come...
BTW, I've removed lots of print statements including those seen by @doutriaux1

**Comment by mcenerney1** [comment]
I'll pull it and give it a try.

**Comment by painter1** [comment]
All is fixed now - at least everything which @mcenerney1 , @doutriaux1 , or I have caught.

**Comment by painter1** [comment]
wait a bit longer.  After running many minutes, your metadiags line had an error, looks easy to fix though

**Comment by mcenerney1** [comment]
When I run metadiags each execution of diags reports an error with "failed with code of 1".
But in fact they do run.  This must be some issue with multiprocessing. I'll look into it.

**Comment by mcenerney1** [comment]
In AMWG_5_NCEP_TS_ANN.log I'm seeing a join issue

Traceback (most recent call last):
  File "/Users/mcenerney1/anaconda/bin/diags", line 762, in <module>
    run_diags(o)
  File "/Users/mcenerney1/anaconda/bin/diags", line 356, in run_diags
    postn=postname, region=r_fname, aux=[aux] )
  File "/Users/mcenerney1/anaconda/lib/python2.7/site-packages/metrics/frontend/form_filenames.py", line 133, in form_file_rootname
    fname = '_'.join( basen, season, '_'.join(vars), '_'.join(aux), postn )

**Comment by painter1** [comment]
The "code 1" is pretty much meaningless so far as I can tell.  The vast majority of the failures are about requested variables which do not exist in the data.  If we use this extensively for testing purposes in the future (and we should), we should clearly separate errors like that from others.

**Comment by painter1** [comment]
@mcenerney1 the log file you saw, AMWG_5_NCEP_TS_ANN.log, refers to obsolete code - line 133 of form_file_rootname no longer looks like that; it calls underscore_join.  Would you please re-run?

**Comment by mcenerney1** [comment]
My hunch is that there is some sort of benign error but the logger is interfering. For now we'll put this on the back burner.

**Comment by mcenerney1** [comment]
The join error is gone.

**Comment by mcenerney1** [comment]
This includes the templatefactory PR also. All tests passed!

**Pull request merged**